src/services/llms/base.py

```diff
@@ -1,13 +1,79 @@
+# src/services/llm/base.py
 from abc import ABC, abstractmethod
-from typing import List, Dict, Any
+import json
+from typing import List, Dict
+from unidiff import Hunk, PatchedFile
 from ...core.models import PRDetails
 
-
 class BaseLLMService(ABC):
+    """
+    Abstract base class for LLM services defining the common interface
+    that all LLM implementations must follow.
+    """
+    
     @abstractmethod
-    def create_prompt(self, file_path: str, diff_content: str, pr_details: PRDetails) -> str:
+    def create_prompt(self, file: PatchedFile, hunk: Hunk, pr_details: PRDetails) -> str:
+        """
+        Create a prompt for the LLM model.
+        
+        Args:
+            file: The file being reviewed
+            hunk: The code hunk to review
+            pr_details: Pull request details
+            
+        Returns:
+            str: Formatted prompt for the LLM
+        """
         pass
-    
+
     @abstractmethod
-    def get_review(self, prompt: str) -> List[Dict[str, Any]]:
+    def get_ai_response(self, prompt: str) -> List[Dict[str, str]]:
+        """
+        Get response from the LLM model.
+        
+        Args:
+            prompt: The formatted prompt to send to the LLM
+            
+        Returns:
+            List[Dict[str, str]]: List of review comments
+        """
         pass
+
+    def _clean_response_text(self, text: str) -> str:
+        """
```

Comment:
The `_clean_response_text` function assumes a specific format (````json````).  This is a tight coupling and might break if the LLM's response format changes. Consider making this function more robust to handle various formats, or at least provide a mechanism to configure the expected format.  Perhaps a configuration parameter to indicate the response delimiters would make it more flexible.